AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2000,6 +2082,37 @@ - (void)layout
     subnode = ((ASDisplayNode *)subnodeLayout.layoutableObject);
     [subnode setFrame:subnodeFrame];
   }
+  
+  if ([[self class] usesImplicitHierarchyManagement]) {
+    if (!_managedSubnodes) {
+      _managedSubnodes = [NSMutableArray array];
+    }
+    
+    for (_ASDisplayNodePosition *position in _deletedSubnodes) {
+      [self _implicitlyRemoveSubnode:position.node atIndex:position.index];
+    }
+    
+    for (_ASDisplayNodePosition *position in _insertedSubnodes) {
+      [self _implicitlyInsertSubnode:position.node atIndex:position.index];
+    }
+  }
+}
+
+- (void)_implicitlyInsertSubnode:(ASDisplayNode *)node atIndex:(NSUInteger)idx
+{
+  ASDisplayNodeAssert(idx <= [_managedSubnodes count], @"index needs to be in range of the current managed subnodes");
```

Comment:
We probably need an assert thread affinity here, as addSubnode on a loaded node must be on main unfortunately...